api/test/common/spinlock_benchmark.cc

```diff
@@ -103,29 +103,17 @@ static void BM_ProcYieldSpinLockThrashing(benchmark::State &s)
 // SpinLock thrashing with thread::yield() after N spins.
 static void BM_ThreadYieldSpinLockThrashing(benchmark::State &s)
 {
-  std::atomic<bool> mutex(false);
-  SpinThrash<std::atomic<bool>>(
+  std::atomic_flag mutex = ATOMIC_FLAG_INIT;
+  SpinThrash<std::atomic_flag>(
       s, mutex,
-      [](std::atomic<bool> &l) {
-        if (!l.exchange(true, std::memory_order_acq_rel))
+      [](std::atomic_flag &l) {
+        while (l.test_and_set(std::memory_order_acq_rel))
         {
-          return;
-        }
-        for (std::size_t i = 0; i < 128; ++i)
-        {
-          if (!l.load(std::memory_order_acquire) && !l.exchange(true, std::memory_order_acq_rel))
-          {
-            return;
-          }
-
-          if (i % 32 == 0)
-          {
-            std::this_thread::yield();
-          }
+          ;
         }
         std::this_thread::yield();
       },
-      [](std::atomic<bool> &l) { l.store(false, std::memory_order_release); });
+      [](std::atomic_flag &l) { l.clear(std::memory_order_release); });
```

Comment:
Thanks for the fix. would you help understand this change (probably enrich PR description) how it is fixing the race condition.

Reply:
I updated the PR description. Please let me know if it needs more information is needed.

Reply:
Thanks for the update. Should we either
 - Change the test comment ( // SpinLock thrashing with thread::yield() after N spins ) as there is no more yield happening.
 - Add the yield after every few tries in the tight loop to be in sync with the benchmark comment.